config/base.yml

```diff
@@ -29,6 +29,13 @@ cache:
     - $HOME/.composer/cache/files
 
 before_script:
+  # Dynamically work out COMPOSER_ROOT_VERSION:
+  # Numeric branch - 4 / 4.8 => 4.x-dev / 4.8.x-dev
+  # Tag - 1.2.3 / 1.2.3-beta1 => 1.2.3 / 1.2.3-beta1
+  # Non-numeric branch - master / some-branch => dev-master / dev-some-branch
+  # $TRAVIS_BRANCH documentation - https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
+  - if [[ "$TRAVIS_BRANCH" =~ ^[1-9]$ ]] || [[ "$TRAVIS_BRANCH" =~ ^[1-9]\.[0-9]+$ ]]; then export COMPOSER_ROOT_VERSION="${TRAVIS_BRANCH}.x-dev"; elif [[ "$TRAVIS_BRANCH" =~ ^[1-9]\.[0-9]+\.[0-9]+ ]]; then export COMPOSER_ROOT_VERSION="${TRAVIS_BRANCH}"; else export COMPOSER_ROOT_VERSION="dev-${TRAVIS_BRANCH}"; fi
```

Comment:
```suggestion
  - if [[ "$TRAVIS_BRANCH" =~ ^[1-9]+$ ]] || [[ "$TRAVIS_BRANCH" =~ ^[1-9]+\.[0-9]+$ ]]; then export COMPOSER_ROOT_VERSION="${TRAVIS_BRANCH}.x-dev"; elif [[ "$TRAVIS_BRANCH" =~ ^[1-9]+\.[0-9]+\.[0-9]+ ]]; then export COMPOSER_ROOT_VERSION="${TRAVIS_BRANCH}"; else export COMPOSER_ROOT_VERSION="dev-${TRAVIS_BRANCH}"; fi
```

Reply:
Yeah I considered that, though, given the highest module we have is userforms at `5`, this will probably take 10 years to show up :-)  I choose to just have a tighter regex instead.

Reply:
:shrug: